docs/static/azure-module.asciidoc

```diff
@@ -0,0 +1,591 @@
+-----
+
+===== Advanced configuration sample
+
+Advanced configuration supports Event Hub specific options. 
+It is not necessary or recommended for most use cases. Use
+it only if it is required for your deployment scenario. 
+
+You must define the `header` array with `name` in the first position. You can
+define other options in any order. The per Event Hub configuration takes
+precedence. Any values not defined per Event Hub use the global config value.  
+
+In this example 'consumer_group' will be applied to each of the configured Event
+Hubs. Note that 'decorate_events' is defined in both the 'global' and per Event Hub
+configuration. The per Event Hub configuration takes precedence, and the
+global configuration is effectively ignored.
+
+["source","shell",subs="attributes"]
+-----
+modules:
+  - name: azure
+    var.elasticsearch.hosts: "localhost:9200"
+    var.kibana.host: "localhost:5601"
+    var.input.azure_event_hubs.threads: 8
+    var.input.azure_event_hubs.decorate_events: true
+    var.input.azure_event_hubs.consumer_group: logstash
+    var.input.azure_event_hubs.event_hubs:
+      - ["name",                      "event_hub_connection",      "storage_connection",                                      "initial_position", "decorate_events"]
+      - ["insights-operational-logs", "Endpoint=sb://example1...", "DefaultEndpointsProtocol=https;AccountName=example1....", "HEAD",             "true"]
+      - ["insights-metrics-pt1m",     "Endpoint=sb://example2...", "DefaultEndpointsProtocol=https;AccountName=example2....", "TAIL",             "true"]
+      - ["insights-logs-errors",      "Endpoint=sb://example3...", "DefaultEndpointsProtocol=https;AccountName=example3....", "TAIL",             "false"]
+      - ["insights-operational-logs", "Endpoint=sb://example4...", "DefaultEndpointsProtocol=https;AccountName=example4....", "HEAD",             "true"]
+-----
+
+[[azure_config_options]]
+===== Configuration options
+
+NOTE: All Event Hubs options are common to both basic and advanced
+configurations, with the following exceptions. The basic configuration uses
+`event-hub-connections`. The the advanced configuration uses `event_hubs` and
+`event_hub_connection`.
+
+[id="plugins-{type}s-{plugin}-config_mode"]
+===== `config_mode`
+* Value type is <<string,string>>
+* Valid entries are `basic` or `advanced`
+* Default value is `basic`
+
+[source,ruby]
+----
+azure_event_hubs {
+   event_hub_connections => ["Endpoint=sb://example1...;EntityPath=event_hub_name1"  , "Endpoint=sb://example2...;EntityPath=event_hub_name2"  ]
+}
+----
+
+[id="plugins-{type}s-{plugin}-event_hubs"]
+===== `event_hubs`
+* Value type is <<array,array>>
+* No default value
+* Ignored for basic configuration
+* Required for advanced configuration 
+
+Defines the Event Hubs to be read. An array of hashes where each entry is a
+hash of the Event Hub name and its configuration options.   
+
+[source,ruby]
+----
+azure_event_hubs {
```

Comment:
we should avoid using the plugin examples.  the module maps the yaml configuration to the plugin configuration. I think the descriptions (for the most part) apply, but the examples, if any provided, should be in the yaml style.
For example:
```
    var.input.azure_event_hubs.event_hubs:
      - ["name",                      "event_hub_connection",      "storage_connection",                                      "initial_position", "decorate_events"]
      - ["insights-operational-logs", "Endpoint=sb://example1...", "DefaultEndpointsProtocol=https;AccountName=example1....", "HEAD",             "true"]
      - ["insights-metrics-pt1m",     "Endpoint=sb://example2...", "DefaultEndpointsProtocol=https;AccountName=example2....", "TAIL",             "true"]
```

Reply:
@jakelandis  Gotcha. 
WRT: "the examples, if any provided..."

Before I do a lot of unnecessary reformatting, let's decide if we need the examples at all.  If we need just a few examples, I'll focus on reworking those.

Reply:
maybe we can provide a an example of how to map the input the options to yaml, and then hyperlink each of the options to the input documentation. Maybe just a bullet list of the linked options ?